### PR TITLE
docs(811): add font metrics calculator to theme docs

### LINKED ISSUE #811: Improve Text cropping Docs
# Description

Anything we could do to improve the docs or storybook around text cropping? eg Point to an old build of the capsize site (or scrape information from that) as the new site does not make as much sense.?

Look at most frequent FAQs for text crop, make sure these are addressed in the docs. 

Relevant Slack Thread - https://newsuktechnology.slack.com/archives/C024ZGQ2W8K/p1680617868065319

# Business Benefit/Value
Help brands like Dow Jones, avoid issues when upgrading from NewsKit 6.

# Acceptance Criteria

Go through the process, proposed update to docs, seek feedback from design 

- Update instructions in - https://newskit.co.uk/theme/theming/creating-a-theme/#createtheme-function , Could we also convert this page to TSX so the deep linking works
- https://newskit.co.uk/theme/foundation/fonts/#textcrop this page should be updated to point to the instructions on the create theme page.

### PR DESCRIPTION
**Pull request by mstuartf**
closes #811 

**What**

1. Background - why this is needed
- Text cropping docs reference an external site (capsize) that we don't control
2. What did you do
- Deleted the creating a theme mdx page and created a tsx one
- Copied the capsize text cropping calculator components across to our repo
- Swapped in Newskit components
- Had to add a simple Select when selecting Google fonts (the capsize docs use a searchable Select with autocomplete, which we don't have out of the box)
- Kept dependencies added to a minimum
3. What does the reviewers should expect
- Interactive text cropping calculator on the creating a theme page
- The page jumps the first time you open a Select. This is a known bug unrelated to this page. 

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/811-text-crop-docs
  Storybook: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/811-text-crop-docs/storybook

File: package.json [hunk 1]
```diff
@@ -224,24 +224,38 @@
   "dependencies": {
     "@babel/runtime-corejs3": "^7.11.2",
     "@capsizecss/core": "^3.0.0",
+    "@capsizecss/metrics": "^1.1.1",
```
Comment (jps) [inline package.json:227]: please don't forget to update `dependenciesNotToBeIncluded` in `scripts/build-package-json.js` so these deps don't get included in the build components package. Ideally we could get to the point where the site and comps are separate packages in the future so this isn't needed.

**Comment by JohnTParsons** [comment]
Sometimes when I use the font Select on http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/811-text-crop-docs/theme/theming/creating-a-theme/, the page jumps to the top, so maybe 'Setting custom font cropping config' should be its own page, rather than deep down on the 'Creating a theme' page.

File: site/public/static/examples/theming/__extends_1.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const stylePresets = {
+  buttonSolidPrimary: Button.stylePresets.buttonSolidPrimary;
+  buttonLightDarkToggle: {
+    __extends: "{{stylePresets.buttonSolidPrimary}}",
```
Comment (mutebg) [inline site/public/static/examples/theming/__extends_1.tsx:4]: I am not sure if we still supports this kind of extend

File: site/pages/theme/theming/creating-a-theme.tsx [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+              inside those brackets are looked up at compile time and replaced
+              with the final value, in this case a hex colour.
+              <br />
+              <br />
+              Tokens can also be used as object keys as well as values. As well
+              as this, multiple tokens can be specified in a single field, this
+              is how the spacing presets are created for example -
+              <InlineCode>
+                {
+                  "spaceInsetSquish030: '{{ sizing.sizing030 }} {{ sizing.sizing040 }}'"
```
Comment (mutebg) [inline site/pages/theme/theming/creating-a-theme.tsx:223]: this is outdated, we don't have `spaceInsetSquish030` tokens anymore

File: site/components/text-crop-calculator/app-state-context.tsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+};
+
+interface StateProviderProps {
+  children: ReactNode;
+}
+
+export function AppStateProvider({children}: StateProviderProps) {
+  const [state, dispatch] = useReducer(reducer, intialState);
+
+  // console.log(state);
```
Comment (mutebg) [inline site/components/text-crop-calculator/app-state-context.tsx:257]: this can be removed

**Comment by mstuartf** [comment]
> Sometimes when I use the font Select on http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/811-text-crop-docs/theme/theming/creating-a-theme/, the page jumps to the top, so maybe 'Setting custom font cropping config' should be its own page, rather than deep down on the 'Creating a theme' page.

@JohnTParsons I left a note about the page jump on the PR description ☝️ 

Re: separate page. I did have that originally but @GeriReid suggested just putting it on the theme page.

**Review by mutebg — APPROVED** [review]
👏  that's awesome, good job

File: site/pages/theme/theming/creating-a-theme.tsx (hunk 3 above)
Comment (mstuartf) [reply site/pages/theme/theming/creating-a-theme.tsx:223]: removed

File: site/public/static/examples/theming/__extends_1.tsx (hunk 2 above)
Comment (mstuartf) [reply site/public/static/examples/theming/__extends_1.tsx:4]: It works ok when defining a theme. What do you mean?

File: site/components/text-crop-calculator/components/output-css.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  return (
+    <Block>
+      <Code>
+        {JSON.stringify(
+          {
+            fontWeight010: 400,
+            fontFamily010: {
+              fontFamily,
+              fontWeight010: {
```
Comment (LukeFinch) [inline site/components/text-crop-calculator/components/output-css.tsx:23]: I think putting this in could be misleading or confusing.
Perhaps need to call out that consumers will need to edit the fontWeight & family when implementing

Comment (mstuartf) [reply site/components/text-crop-calculator/components/output-css.tsx:23]: Good point. Do you have a suggestion on wording?

**Review by JohnTParsons — APPROVED** [review]

**Review by JohnTParsons — DISMISSED** [review]

**Review by mutebg — DISMISSED** [review]

**Review by mutebg — DISMISSED** [review]

**Review by JohnTParsons — APPROVED** [review]

**Review by JohnTParsons — APPROVED** [review]

**Pull request merged**